src/feature/relay/router.c

```diff
@@ -1939,6 +1939,12 @@ get_my_declared_family(const or_options_t *options)
   return result;
 }
 
+#ifdef HAVE_MODULE_RELAY
+#define IPV6_DNS_NOT_BROKEN !dns_seems_to_be_broken_for_ipv6()
```

Comment:
There could be a critical bug here:

dns_seems_to_be_broken_for_ipv6() checks dns_is_broken_for_ipv6. But the patch adds dns_was_broken_for_ipv6, so it looks like you meant to use dns_was_broken_for_ipv6.

Good unit tests would have caught this bug. Better variable naming might also have avoided it.

I also don't think dns_was_broken_for_ipv6 will need a macro, because it's only defined in this file.

Reply:
Please remove the macro for IPV6_DNS_NOT_BROKEN.

It's confusing to hide functions behind a macro. And double negatives are also confusing.

Reply:
Instead, put a function-like macro in the !defined(HAVE_MODULE_RELAY) section of dns.h.